apis/v1alpha2/validation/gateway.go

```diff
@@ -55,8 +82,12 @@ func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path
 func validateListenerHostname(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
 	for i, h := range listeners {
+		tcpUdpProtocols := []gatewayv1a2.ProtocolType{gatewayv1a2.TCPProtocolType,gatewayv1a2.UDPProtocolType}
 		// When unspecified, “”, or *, all hostnames are matched.
 		if h.Hostname == nil || (*h.Hostname == "" || *h.Hostname == "*") {
+			if contains(tcpUdpProtocols,h.Protocol) {
+				errs = append(errs, field.Required(path.Index(i).Child("hostname"), fmt.Sprintf("hostname is required for protocol type %v",h.Protocol)))
+			}
```

Comment:
This is in a block where hostname is effectively empty. Instead we'd want to move this below.

Reply:
The validation we actually want here is to say that Hostname must be empty/nil if the protocol is TCP or UDP.